src/test/java/no/nb/nna/veidemann/db/RethinkAstDecompilerTest.java

```diff
@@ -0,0 +1,90 @@
+        assertThat(new RethinkAstDecompiler(ast).toString()).isEqualTo(expected);
+
+        ast = r.table("config").filter(p1 -> p1.eq("foo").or(p1.eq("bar")).and(p1.ne("aaa")));
+        expected = "r.table(\"config\").filter(p1 -> p1.eq(\"foo\").or(p1.eq(\"bar\")).and(p1.ne(\"aaa\")))";
+        assertThat(new RethinkAstDecompiler(ast).toString()).isEqualTo(expected);
+
+        ast = r.table("config").filter(p2 -> r.args(p2)).orderBy("foo", r.desc(p1 -> p1), r.array("a", "b"));
+        expected = "r.table(\"config\").filter(p2 -> r.args(p2)).orderBy(\"foo\", r.desc(p1 -> p1), r.array(\"a\", \"b\"))";
+        assertThat(new RethinkAstDecompiler(ast).toString()).isEqualTo(expected);
+
+        ast = r.table("config").filter(p1 -> p1.g("kind").eq("crawlScheduleConfig")
+                .and(r.expr(r.array(r.hashMap("key", "foo").with("value", "bar"), r.hashMap("key", "aaa").with("value", "bbb")))
+                        .contains(p1.g("meta").g("label"))));
+        expected = "r.table(\"config\").filter(p1 -> p1.g(\"kind\").eq(\"crawlScheduleConfig\")" +
+                ".and(r.array(r.hashMap(\"key\", \"foo\").with(\"value\", \"bar\"), r.hashMap(\"key\", \"aaa\").with(\"value\", \"bbb\"))" +
+                ".contains(p1.g(\"meta\").g(\"label\"))))";
+        assertThat(new RethinkAstDecompiler(ast).toString()).isEqualTo(expected);
+
+        String t = "2020-12-02T10:13:49.025+01";
+        ast = r.table("mintabell").filter(p1 -> p1.g("foo").contains(r.now(), OffsetDateTime.parse(t),
+                r.time(2020, 12, 2, 10, 13, 49, 25), r.iso8601(p1)));
+        expected = "r.table(\"mintabell\").filter(p1 -> p1.g(\"foo\").contains(r.now(), " +
+                "r.iso8601(\"2020-12-02T10:13:49.025+01\"), r.time(2020, 12, 2, 10, 13, 49, 25), r.iso8601(p1)))";
+        assertThat(new RethinkAstDecompiler(ast).toString()).isEqualTo(expected);
+    }
+
+    @Test
+    void testEquals() {
+        ReqlAst ast1 = r.table("mintabell").getAll("foo", "bar").filter((var1) -> var1.eq(new String[]{"ugh"}));
+        ReqlAst ast2 = r.table("mintabell").getAll("foo", "bar").filter((row) -> row.eq(r.array("ugh")));
+        String expected = "r.table(\"mintabell\").getAll(\"foo\", \"bar\").filter(p1 -> p1.eq(r.array(\"ugh\")))";
+
+        assertThat(new RethinkAstDecompiler(ast1)).isEqualTo(new RethinkAstDecompiler(ast2));
+        assertThat(new RethinkAstDecompiler(ast1).toString()).isEqualTo(expected);
+        assertThat(new RethinkAstDecompiler(ast2).toString()).isEqualTo(expected);
+    }
+
+    @Test
+    void isEqual() {
+        ReqlAst ast1 = r.table("mintabell").getAll("foo", "bar").filter((var1) -> var1.eq(new String[]{"ugh"}));
+        ReqlAst ast2 = r.table("mintabell").getAll("foo", "bar").filter((row) -> row.eq(r.array("ugh")));
+        assertThat(RethinkAstDecompiler.isEqual(ast1, ast2)).isTrue();
+    }
+}
```

Comment:
No newline at end of file
```suggestion
}

```